lumicks/pylake/kymotracker/kymotracker.py

```diff
@@ -367,7 +367,7 @@ def track_lines(
     return refine_tracks_centroid(kymotrack_group, track_width=line_width, bias_correction=True)
 
 
-def filter_tracks(tracks, minimum_length):
+def filter_tracks(tracks, *args, minimum_duration=0, **kwargs):
```

Comment:
hmm, doing it this way I think is not the best option (although I get why you did, so we can throw the `FutureWarning`). It hit me right away because `minimum_length` isn't in the signature, but it is in the docstring. This leads to a very strange API stub:

<img width="726" alt="image" src="https://github.com/lumicks/pylake/assets/61475504/80051ea3-6943-4b48-ad00-f417a70ab463">

I thought there might be a better way using `inspect` but unfortunately it doesn't tell you if an argument was called as positional or keyworded 😑 So I think we're left with a pretty ugly option, but user facing I think is better; something like

```
def filter_tracks(tracks, *args, minimum_length=None, minimum_duration=0):
    if len(args) > 1:
        raise ValueError("Passed too many positional arguments")
    elif len(args):
        if minimum_length is not None:
            raise ValueError("Passed too many positional arguments")
        else:
            minimum_length = args[0]
    minimum_length = 1 if minimum_length is None else minimum_length
```
It's still not optimal because you end up with the random `*args` but at least `minimum_length` is in the signature and the rendered API stub will look better.


An alternative that just came to mind (especially since going this route is going to be ugly no matter what) is to just completely deprecate `lk.filter_tracks()` and add a method `KymoTrackGroup.filter(self, *, minimum_length=1, minimum_duration=0)` which would return a new filtered group instance. In a way, I even think this option more closely matches our general API (we often have mutation methods that return a new instance rather than free functions) and `lk.filter_tracks()` already expects a group instance first anyways.

What do you think?

Reply:
Yeah, that was indeed the reason. The first option doesn't let us get rid of the positional option over time. Then again, I suppose we never really want to add additional positional arguments to this one, so that may be OK.

I don't know if I'm a huge fan of removing it and adding it to `KymoTrackGroup`. I think quite a few (if not most) of the methods on groups behave like the methods you'd expect on a `list` (which is mutate the object rather than return a new instance). If you look at `remove_tracks_in_rect`, `extend`, `split`, `merge`, they all mutate.

Reply:
yeah, I was referring to the API outside `KymoTrackGroup` we generally don't mutate in place. But since we have that behavior anyway we could still do `KymoTrackGroup.filter()` and have it change in place. I guess any option is fine; I'm not particularly a fan of the random `*args` but 🤷 not the worst thing in the world. 

I guess the only other thing with the first option is it doesn't really follow our normal deprecation method, but again not a huge deal. It would be nice to somehow mark in the docstring prominently that the positional argument will be removed in the future. Not sure if there's a nice way to do that. And of course, we need to not forget to actually remove this at some point

Reply:
No, I meant just leaving the argument positional, giving it a default and dropping the `*args`. Essentially dropping all the funky input logic we have now for the free function and accepting that that one argument can just be passed in a positional manner. Maybe that is the best option. We can still always provide it with a named argument in the docs to at least guide users into that habit.

I can add a separate `.filter` on the group and do `kwargs` from the get go there, but I think the free function has value of its own (since copying prior to mutation is annoying). It's kind of like having `np.reshape` vs `array.reshape`.